pkg/api/defaults.go

```diff
@@ -105,7 +105,20 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpdate bool) {
 
 		if o.KubernetesConfig.KubernetesImageBase == "" {
 			o.KubernetesConfig.KubernetesImageBase = cloudSpecConfig.KubernetesSpecConfig.KubernetesImageBase
+
+			// Only copy overrides when the user did not specify an image base for back-compat
+			//
+			// TODO: inject overrides when the user specified overrides but we have overrides
+			// that the user does not?
+			// TODO: Should we just always do this even if the user specified an image base?
```

Comment:
Looks like maybe on upgrade overrides are not applied because the derived model from the original deploy has "KubernetesImageBase" set?

Reply:
Just had a thought, maybe overrides are keyed by image reference instead of component name.
e.g. instead of `{"hyperkube": {// ovrerride }`, do `{"k8s.gcr.io/hyperkube-amd64": { // override }}`

This would require overrides to be more explicit and per cloud, but I think that's generally what we want?